distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/MachineDependentHeap.java

```diff
@@ -142,7 +142,7 @@ public static MachineNodeRole parse(InputStream config) {
                         return MachineNodeRole.DATA;
                     } else if (containsOnly(roles, "master")) {
                         return MachineNodeRole.MASTER_ONLY;
-                    } else if (containsOnly(roles, "ml")) {
+                    } else if (containsOnly(roles, "ml") || containsOnly(roles, "ml", "remote_cluster_client")) {
```

Comment:
This can be simplified to `else if (containsOnly(roles, "ml", "remote_cluster_client")`. That simply asserts that only the listed roles area llowd to be in the given collection, but a subset is fine as well.

Reply:
Thinking now, this would also be `true` in a case where a node had _only_ the `remote_cluster_client` role, so maybe we might want to assert `roles.contains("ml") && containsOnly(roles, "ml", "remote_cluster_client")`

Reply:
Thanks.  I did that and also added a test to prove `remote_cluster_client` on its own isn't an ML node.